src/Core/src/AbstractApi.php

```diff
@@ -85,7 +85,9 @@ public function __construct($configuration = [], ?CredentialProvider $credential
         $request = $input->request();
         $request->setEndpoint($this->getEndpoint($request->getUri(), $request->getQuery()));
 
-        $this->getSigner()->presign($request, $this->credentialProvider->getCredentials($this->configuration), $expires ? $expires : new \DateTimeImmutable('+60min'));
+        if (null !== $credentials = $this->credentialProvider->getCredentials($this->configuration)) {
+            $this->getSigner()->presign($request, $credentials, new RequestContext(['expirationDate' => $expires]));
```

Comment:
Will this work? We dont have operation here.

Reply:
the operation is not a strong requirement, it avoid computing md5 on some S3 endpoint.

Reply:
Yes. But can I presign those endpoints if I dont pass "operation"?

Reply:
Yes it will work.
Because body is not used to compute the signature (special case for S3 using a special hash `UNSIGNED-PAYLOAD`) we don't have have to know the operation, because we don't have to MD5 the body

Reply:
Okey. I understand. 
Thank you